library/pk.c

```diff
@@ -124,13 +133,19 @@ MBEDTLS_ALWAYS_INLINE static inline int pk_info_encrypt_func(
     int (*f_rng)(void *, unsigned char *, size_t),
     void *p_rng )
 {
+    if( info->encrypt_func == NULL )
+        return( MBEDTLS_ERR_PK_TYPE_MISMATCH );
+
     return( info->encrypt_func( ctx, input, ilen, output, olen, osize,
                                 f_rng, p_rng ) );
 }
 
 MBEDTLS_ALWAYS_INLINE static inline int pk_info_check_pair_func(
     const mbedtls_pk_info_t *info, const void *pub, const void *prv )
 {
+    if( info->check_pair_func == NULL )
```

Comment:
Pre-existing: It's not clear to me why we're returning `MBEDTLS_ERR_PK_BAD_INPUT_DATA` here, ubt `MBEDTLS_ERR_PK_TYPE_MISMATCH` elsewhere.

Reply:
Yes, it's not very consistent, but I chose to minimise changes in this PR, so kept the existing inconsistency. In practice I suspect people just check if the value is zero or not, anyway.